service/spring.gradle

```diff
@@ -1,28 +1,22 @@
 configurations { compileOnly { extendsFrom annotationProcessor } }
 
-var springBootVersion = '2.7.4'
+var springBootVersion = '3.1.2'
 
 dependencies {
+	implementation "org.springframework.boot:spring-boot-starter-actuator:$springBootVersion"
 	implementation "org.springframework.boot:spring-boot-starter-web:$springBootVersion"
 	developmentOnly "org.springframework.boot:spring-boot-devtools:$springBootVersion"
 	annotationProcessor "org.springframework.boot:spring-boot-configuration-processor:$springBootVersion"
 	testImplementation "org.springframework.boot:spring-boot-starter-test:$springBootVersion"
 	implementation "org.springframework.boot:spring-boot-starter-thymeleaf:$springBootVersion"
+	// Needed for OpenTelemetry
+	implementation "org.springframework.data:spring-data-commons:$springBootVersion"
 
-	implementation 'org.apache.httpcomponents:httpclient:4.5.13'
+	implementation 'org.apache.httpcomponents.client5:httpclient5'
 	implementation 'org.apache.commons:commons-collections4:4.4'
 
 	// allows mocking final classes
 	testImplementation 'org.mockito:mockito-inline:5.2.0'
-
-	// See https://nvd.nist.gov/vuln/detail/CVE-2023-46589
-	// TODO DR-3393: remove this pin when upgrading Spring Boot, TCL
```

Comment:
No longer needed (and in fact, this earlier pin was incompatible with other upgrades):

```
(base) okotsopo@wm111-e35 terra-drs-hub % ./gradlew service:dependencyInsight --dependency tomcat-embed-core

> Task :service:dependencyInsight
org.apache.tomcat.embed:tomcat-embed-core:10.1.11 (selected by rule)
  Variant compile:
    | Attribute Name                 | Provided | Requested    |
    |--------------------------------|----------|--------------|
    | org.gradle.status              | release  |              |
    | org.gradle.category            | library  | library      |
    | org.gradle.libraryelements     | jar      | classes      |
    | org.gradle.usage               | java-api | java-api     |
    | org.gradle.dependency.bundling |          | external     |
    | org.gradle.jvm.environment     |          | standard-jvm |
    | org.gradle.jvm.version         |          | 17           |

org.apache.tomcat.embed:tomcat-embed-core:10.1.11
+--- org.apache.tomcat.embed:tomcat-embed-websocket:10.1.11
|    \--- org.springframework.boot:spring-boot-starter-tomcat:3.1.2
|         \--- org.springframework.boot:spring-boot-starter-web:3.1.2
|              \--- compileClasspath
\--- org.springframework.boot:spring-boot-starter-tomcat:3.1.2 (*)

(*) - dependencies omitted (listed previously)
```